src/Util/ConfigFileLoader.php

```diff
@@ -56,41 +109,68 @@ public function setupCache(ConfigCache $config, $raw = false)
 	}
 
 	/**
-	 * Tries to load the specified core-configuration and returns the config array.
+	 * Tries to load the static core-configuration and returns the config array.
 	 *
-	 * @param string $name The name of the configuration (default is empty, which means 'local')
+	 * @param string $name The name of the configuration
 	 *
 	 * @return array The config array (empty if no config found)
 	 *
-	 * @throws \Exception if the configuration file isn't readable
+	 * @throws Exception if the configuration file isn't readable
 	 */
-	public function loadCoreConfig($name = '')
+	private function loadStaticConfig($name)
 	{
-		if (!empty($this->getConfigFullName($name))) {
-			return $this->loadConfigFile($this->getConfigFullName($name));
-		} elseif (!empty($this->getIniFullName($name))) {
-			return $this->loadINIConfigFile($this->getIniFullName($name));
+		$configName = $this->staticDir . DIRECTORY_SEPARATOR . $name . '.config.php';
+		$iniName    = $this->staticDir . DIRECTORY_SEPARATOR . $name . '.ini.php';
+
+		if (file_exists($configName)) {
+			return $this->loadConfigFile($configName);
+		} elseif (file_exists($iniName)) {
+			return $this->loadINIConfigFile($iniName);
 		} else {
 			return [];
 		}
 	}
 
+	/**
+	 * Tries to load the specified core-configuration into the config cache.
+	 *
+	 * @param ConfigCache $config The Config cache
+	 *
+	 * @return array The config array (empty if no config found)
+	 *
+	 * @throws Exception if the configuration file isn't readable
+	 */
+	private function loadCoreConfig(ConfigCache $config)
+	{
+		// try to load legacy ini-files first
+		foreach ($this->getConfigFiles(true) as $configFile) {
+			$config->load($this->loadINIConfigFile($configFile), true);
+		}
+
+		// try to load supported config at last to overwrite it
+		foreach ($this->getConfigFiles() as $configFile) {
+			$config->load($this->loadConfigFile($configFile), true);
+		}
+
+		return [];
+	}
+
 	/**
 	 * Tries to load the specified addon-configuration and returns the config array.
 	 *
 	 * @param string $name The name of the configuration
 	 *
 	 * @return array The config array (empty if no config found)
 	 *
-	 * @throws \Exception if the configuration file isn't readable
+	 * @throws Exception if the configuration file isn't readable
 	 */
 	public function loadAddonConfig($name)
 	{
 		$filepath = $this->baseDir . DIRECTORY_SEPARATOR . // /var/www/html/
-			Addon::DIRECTORY       . DIRECTORY_SEPARATOR . // addon/
-			$name                  . DIRECTORY_SEPARATOR . // openstreetmap/
-			self::SUBDIRECTORY     . DIRECTORY_SEPARATOR . // config/
-			$name . ".config.php";                         // openstreetmap.config.php
+					Addon::DIRECTORY . DIRECTORY_SEPARATOR . // addon/
```

Comment:
Please restore the previous formatting. Also as a rule of thumb, `DIRECTORY_SEPARATOR` is useless when the produced path is directly read by PHP. Windows platforms support file paths with forward slashes, so the `DIRECTORY_SEPARATOR` is only useful for outputting a familiar display to users between platforms.

Reply:
I'm not what you mean to be honest ... I thought `DIRECTORY_SEPARATOR` is necessary to make sure that the path is readable on linux and windows plattforms ..

Reply:
Nope, `path/to/file` is equally readable by PHP either on Linux, MacOS or Windows. In practice you can use forward slash for everything but path display because `path/to/file` may confuse Windows users. I think even Windows Explorer accepts forward slashes in the address bar.

Reply:
Good to know!